UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -3093,135 +3095,156 @@ public async void UpdateApp(SettingsWindow window)
             };
             SetProgressBar();
 
-            using (WebClient webClient = new())
+            try
             {
-                bool end = false;
-                bool ended = false;
-                string downloaded = "0.00";
-
-                webClient.DownloadProgressChanged += new DownloadProgressChangedEventHandler((sender, e) =>
-                {
-                    if (!end)
-                        downloaded = (e.BytesReceived / bytesToMB).ToString("F2", CultureInfo.InvariantCulture);
-                });
-                webClient.DownloadFileCompleted += new AsyncCompletedEventHandler((sender, e) =>
+                _ = Task.Run(async () =>
                 {
-                    end = true;
-
-                    HideProgressBar();
-                    _ = Task.Run(() =>
-                    {
-                        // wait until progress bar updater loop is finished
-                        while (!ended)
-                            Thread.Sleep(100);
-
-                        scriptDialog = null;
-                    });
-
-                    if (e.Error is not null)
+                    using (HttpClient httpClient = new() { Timeout = TimeSpan.FromHours(1) })
```

Comment:
can you change this to half an hour or 20 min instead?
1H is a waaay to big timeout IMO. Even if you have a worst case of our builds being 200MB and someone only downloading 200kb per second, thatd still be done in ~15min.

Or is this just for the download request? in which case it should definitely not be longer than 5min